frost-p256/src/lib.rs

```diff
@@ -0,0 +1,275 @@
+#![allow(non_snake_case)]
+#![deny(missing_docs)]
+#![doc = include_str!("../README.md")]
+
+use elliptic_curve::hash2curve::{hash_to_field, ExpandMsgXmd};
+use hex::FromHex;
+use p256::{
+    elliptic_curve::{
+        sec1::{FromEncodedPoint, ToEncodedPoint},
+        Field as FFField, PrimeField,
+    },
+    AffinePoint, ProjectivePoint, Scalar,
+};
+use rand_core::{CryptoRng, RngCore};
+use sha2::{digest::Update, Digest, Sha256};
+
+use frost_core::{frost, Ciphersuite, Field, Group};
+
+#[cfg(test)]
+mod tests;
+
+pub use frost_core::Error;
+
+#[derive(Clone, Copy)]
+/// An implementation of the FROST ciphersuite scalar field.
+pub struct P256ScalarField;
+
+impl Field for P256ScalarField {
+    type Scalar = Scalar;
+
+    type Serialization = [u8; 32];
+
+    fn zero() -> Self::Scalar {
+        Scalar::ZERO
+    }
+
+    fn one() -> Self::Scalar {
+        Scalar::ONE
+    }
+
+    fn invert(scalar: &Self::Scalar) -> Result<Self::Scalar, Error> {
+        // [`curve25519_dalek::scalar::Scalar`]'s Eq/PartialEq does a constant-time comparison using
+        // `ConstantTimeEq`
```

Comment:
We should update this comment after investigating what `p256::Scalar` does under the hood for `Eq`, `PartialEq` for comparison (pretty sure it does the right thing just making sure)

Reply:
Oops, done in 2fb7adc. I double checked, it does the same.

Would it make sense to add a trait bound to `ConstantTimeEq` in `frost-core`?

Reply:
It's not a strict requirement, more like a nice to have. I don't necessarily want to put that constraint on library users, as it's a third-party trait and not a `std` or `core` trait, even if all the nice libraries _we_ plan to use to implement FROST with support `ConstantTimeEq`, like https://docs.rs/k256/latest/k256/struct.Scalar.html